### DISCUSSION TITLE
How do I get this project running?

### DISCUSSION THREAD

**Discussion by mio4kon** [agoda-com/macOS-vz-kubelet#3] [Q&A]
Do I first need to add the macOS machine as a node to the Kubernetes cluster and then operate it? I didn’t quite understand the README, mainly because I'm not very familiar with virtual-kubelet.

**Comment by Raikerian** [comment]
Hi. Virtual Kubelet is just an agent, so you need an already running Kubernetes cluster. After that get your k8s apiserver certs and pass them through env variables listed as required in the usage guide https://github.com/agoda-com/macOS-vz-kubelet?tab=readme-ov-file#usage-guide
Upon running the application, it will automatically register itself as a node in your target cluster and the rest of the flow will be available for you.
